src/platform/nm-platform-utils.c

```diff
@@ -384,6 +384,57 @@ nmp_utils_ethtool_get_link_settings (int ifindex,
 	return TRUE;
 }
 
+
+#define ADVERTISED_INVALID 0
+#define BASET_ALL_MODES (  ADVERTISED_10baseT_Half \
+                         | ADVERTISED_10baseT_Full \
+                         | ADVERTISED_100baseT_Half \
+                         | ADVERTISED_100baseT_Full \
+                         | ADVERTISED_1000baseT_Half \
+                         | ADVERTISED_1000baseT_Full \
+                         | ADVERTISED_10000baseT_Full )
+
+static inline unsigned long int
+get_baset_mode (guint32 speed, NMPlatformLinkDuplexType duplex)
+{
+	int speed_index, duplex_index;
+	const unsigned long int baset_modes[2][4] = { { ADVERTISED_10baseT_Half,
+	                                                ADVERTISED_100baseT_Half,
+	                                                ADVERTISED_1000baseT_Half,
+	                                                ADVERTISED_INVALID },
+	                                              { ADVERTISED_10baseT_Full,
+	                                                ADVERTISED_100baseT_Full,
+	                                                ADVERTISED_1000baseT_Full,
+	                                                ADVERTISED_10000baseT_Full } };
+
+	if (duplex == NM_PLATFORM_LINK_DUPLEX_UNKNOWN)
+		return ADVERTISED_INVALID;
+
+	if (duplex == NM_PLATFORM_LINK_DUPLEX_HALF)
+		duplex_index = 0;
+	else
+		duplex_index = 1;
+
+	switch (speed) {
+		case 10:
+			speed_index = 0;
+			break;
+		case 100:
+			speed_index = 1;
+			break;
+		case 1000:
+			speed_index = 2;
+			break;
+		case 10000:
+			speed_index = 3;
+			break;
+		default:
+			return ADVERTISED_INVALID;
+			break;
+	}
```

Comment:
wouldn't it be simpler to just return the selected value right away?
```
if (duplex == NM_PLATFORM_LINK_DUPLEX_HALF) {
    switch (speed) {
        case 10: return ADVERTISED_10baseT_Half;
        ...
        default: return ADVERTISED_INVALID;
    }
} else {
    ...
```

A lookup array makes sense, if you can avoid the if-else search for the index. But you don't avoid that. It only adds more code.

Reply:
Didn't liked the double switch construct... but well, I agree the alternative didn't looked really better. Maybe a little worse as less straightforward.
So, switched to plain construct as you proposed.